core/modules/search/search.admin.inc

```diff
@@ -105,66 +119,83 @@ function search_admin_settings($form, &$form_state) {
-  );
-
-
-  // Collect some stats
-  $remaining = 0;
-  $total = 0;
-  foreach ($config->get('search_active_modules') as $module) {
-    if ($status = module_invoke($module, 'search_status')) {
-      $remaining += $status['remaining'];
-      $total += $status['total'];
+  // Display node-related configurations only if content search is enabled.
+  if (in_array('node', $config->get('search_active_modules'), TRUE)) {
+    // Indexing settings:
+    $form['indexing'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Indexing'),
+      '#description' => t('<p><em>The search index is not cleared immediately when invalidated, instead it will be systematically updated during cron runs. Searching will continue to work but new content won\'t be indexed until all existing content has been re-indexed.</em></p>'),
+    );
+    // Collect indexing stats per content type.
+    $report = '';
+    foreach ($content_types as $type => $search) {
+      if ($search == TRUE) {
+        $total = db_query('SELECT COUNT(*) FROM {node} WHERE type = :type', array(':type' => $type))->fetchField();
+        $remaining = db_query("SELECT COUNT(*) FROM {node} n LEFT JOIN {search_dataset} d ON d.type = 'node' AND d.sid = n.nid WHERE (d.sid IS NULL OR d.reindex <> 0) AND n.type = :type", array(':type' => $type))->fetchField();
+        $count = format_plural($remaining, '<strong><em>1</em></strong> node of the content type left to index or excluded from indexing.', '<strong><em>@count</em></strong> nodes of the content type left to index or excluded from indexing.');
+        $percentage = ((int) min(100, 100 * ($total - $remaining) / max(1, $total))) . '%';
+        // Finally output all the stats.
+        $report .= '<p>' . t(
+          '<strong>%percentage</strong> of total <strong>%total</strong>  <em>@content-type</em> type nodes have been indexed.',
+          array(
+          '%percentage' => $percentage,
+          '%total' => $total,
+          '@content-type' => $node_type_get_types[$type]->name,
+        )
+        )
+        . ' ' . $count . '</p>';
+      }
     }
+    // Collect total indexing stats.
+    $remaining = 0;
+    $total = 0;
+    foreach ($config->get('search_active_modules') as $module) {
+      if ($status = module_invoke($module, 'search_status')) {
+        $remaining += $status['remaining'];
+        $total += $status['total'];
+      }
+    }
+    $remaining_count = format_plural($remaining, '<strong>1</strong> node left to index or excluded from indexing.', '<strong>@count</strong> nodes left to index or excluded from indexing.');
+    $percentage = ((int) min(100, 100 * ($total - $remaining) / max(1, $total))) . '%';
+    $report .= '<p>' . t('<strong>%percentage</strong> of total <strong>%total</strong> nodes have been indexed.', array('%percentage' => $percentage, '%total' => $total)) . ' ' . $remaining_count . '</p>';
+    $form['indexing']['report'] = array('#markup' => $report);
```

Comment:
From here on down, this form should not be wrapped in the `if (in_array('node', $config->get('search_active_modules'), TRUE)) {` statement. Invalidating search index, the search cron limit, and other options are not dependent on having node search enabled.